lib/hal_client/link.rb

```diff
@@ -56,7 +56,7 @@ def self.new_from_link_entry(options)
 
       rel = hash_entry[:rel]
       hash_data = hash_entry[:data]
-      href = (base_url + hash_data['href']).to_s
+      href = (base_url + hash_data['href'].to_s).to_s
```

Comment:
The target of the link should be nil if the href is nil. This will make the target of a link the base url, right?